### DISCUSSION TITLE
Why are there two JS files now being output on build?

### DISCUSSION THREAD

**Discussion by Cristy94** [parcel-bundler/parcel#9074] [Q&A] [answered]
Previously only a JS minified artifact will be created and linked from index.html, now it looks like two of them are being created (a normal one and a really tiny one):
<img width="554" alt="image" src="https://github.com/parcel-bundler/parcel/assets/1384885/3f894170-de64-473c-9252-f3052c4e93c9">

When did this change happen? Why are there two files now?

**Comment by mischnic** [comment]
Please provide a reproduction

**Reply by Cristy94** [reply]
Wait, so this is not an intended/known change?

The project is really big, I'll see if I can create a smaller version showing the issue.

The thing is that the project didn't change, just upgrading the Parcel version lead to two bundles being now created.

**Comment by mischnic** [comment]
If you don't have dynamic imports and only a single HTML script tag, then no it shouldn't happen.

**Reply by Cristy94** [reply]
Hmm, before, there was a problem with the CSS insert order, which I fixed using an empty script tag:

https://github.com/parcel-bundler/parcel/issues/4882#issuecomment-872879634

<img width="1195" alt="image" src="https://github.com/parcel-bundler/parcel/assets/1384885/77151d62-17a7-438a-b49b-bff77f534225">

Though, after removing the empty script tag, there are still two bundles being created:

<img width="632" alt="image" src="https://github.com/parcel-bundler/parcel/assets/1384885/406327b8-8ba4-4106-a965-34b63535f182">
<img width="563" alt="image" src="https://github.com/parcel-bundler/parcel/assets/1384885/648e2ba2-32c6-4939-b52b-dd14048264e6">

**Reply by Cristy94** [reply]
Is there any way to enable some Debug mode to see why that file is being created?

**Reply by Cristy94** [reply]
Actually, something is way off. Not only that, but it looks like  the build process is being ran twice?

<img width="524" alt="image" src="https://github.com/parcel-bundler/parcel/assets/1384885/1b303a9d-e24f-4704-ba1c-8e410340c6f3">


<img width="1148" alt="image" src="https://github.com/parcel-bundler/parcel/assets/1384885/5f640695-6cc6-4aac-a1de-843a4ba5aeb1">

**Reply by mischnic** [reply]
Ah, another reason would be that a module script tag in the source becomes in the output a module plus a nomodule script tag (with corresponding js bundles) if your browserslist config contains browser not supporting module scripts.

**Reply by Cristy94** [reply]
"browserslist": [
    "last 1 version",
    "> 1%",
    "not dead"
  ],

**Reply by Cristy94** [reply]
@mischnic Ok, so changing to "last 1 Chrome version"  fix the double-build issue and only builds once, but still outputs 2 JS files (big one plus the small one below).

**Comment by Cristy94** [comment]
If it helps, here is the content of the small file:

```
function e(e,r,n,o){Object.defineProperty(e,r,{get:n,set:o,enumerable:!0,configurable:!0})}var r="undefined"!=typeof globalThis?globalThis:"undefined"!=typeof self?self:"undefined"!=typeof window?window:"undefined"!=typeof global?global:{},n={},o={},t=r.parcelRequire51ce;null==t&&((t=function(e){if(e in n)return n[e].exports;if(e in o){var r=o[e];delete o[e];var t={id:e,exports:{}};return n[e]=t,r.call(t.exports,t,t.exports),t.exports}var f=Error("Cannot find module '"+e+"'");throw f.code="MODULE_NOT_FOUND",f}).register=function(e,r){o[e]=r},r.parcelRequire51ce=t),t.register("jXExt",function(r,n){"use strict";e(r.exports,"register",function(){return o},function(e){return o=e}),e(r.exports,"resolve",function(){return t},function(e){return t=e});var o,t,f={};o=function(e){for(var r=Object.keys(e),n=0;n<r.length;n++)f[r[n]]=e[r[n]]},t=function(e){var r=f[e];if(null==r)throw Error("Could not resolve bundle with id "+e);return r}}),t("jXExt").register(JSON.parse('{"GMSaN":"index.4c90c207.js","3fZvm":"click.05c9d4ca.mp3","7lCEH":"cursor.a11d7ca9.png","5Ycmz":"logo-small.ed83391b.png","6lpD3":"chrome.c79d8d9a.png","cM3x1":"opera.c890b9c9.png","jzBhz":"ie.b1844920.png","J6RgF":"firefox.b4cdc80e.png","500Ae":"safari.11b804b2.png","5wwDY":"mobile.cd1eaa44.png","hQ5sO":"index.f7b4cd15.css"}'));
```

**Reply by Cristy94** [reply]
Seems like it registers/loads some static assets (?).

**Reply by Cristy94** [reply]
Another fun fact, the big one is included like this:

```<script src="index.0df5d241.js" type="module">```

While the small one:

```<script type="module" src="index.80f5b43e.js">```

But it might just be the minifier changing the order of the script attributes.

**Reply by mischnic** [reply, marked as answer]
This should be caused by https://github.com/parcel-bundler/parcel/pull/8837

**Reply by Cristy94** [reply]
This comment from there would have saved me some time:

> I think it'd be useful to name these bundles so we know what they are (e.g. runtime-manifest.HASH.js). 

How do I disable the split feature?

Should I just pass `--splitManifestThreshold 9999999` to build ?

**Reply by Cristy94** [reply]
splitManifestThreshold  Doesn't seem to be a build option.
